### ISSUE TITLE
`CA1066` triggers incorrectly on `struct System.IntPtr : IEquatable<nint>`

### ISSUE THREAD

**Issue by alexrp** [dotnet/roslyn-analyzers#5354]
### Analyzer

**Diagnostic ID**: [CA1066](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1066): `Implement IEquatable when overriding Equals`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: `6.0.100-preview.6.21355.2`

### Describe the bug

`CA1066` triggers on:

```csharp
namespace System
{
    public readonly struct IntPtr : IEquatable<nint>
    {
        // ...
    }
}
```

Changing `IEquatable<nint>` to `IEquatable<IntPtr>` makes the warning go away.

### Expected behavior

No warning. `IntPtr` and `nint` should be equivalent.

### Actual behavior

`CA1066` is triggered.

## Additional context

This happens in a custom `System.Private.CoreLib` implementation.

**Comment by alexrp** [comment]
The same happens for `UIntPtr`/`nuint`.
